File: app/in/partake/controller/api/debug/SuccessIfLoginAPI.java
package in.partake.controller.api.debug;

import play.mvc.Result;
import in.partake.base.PartakeException;
import in.partake.controller.api.AbstractPartakeAPI;
import in.partake.model.UserEx;
import in.partake.model.dao.DAOException;

public class SuccessIfLoginAPI extends AbstractPartakeAPI {

    public static Result get() throws DAOException, PartakeException {
        return new SuccessIfLoginAPI().execute();
    }

    /**
     * login していれば success と同じ挙動をする。
     * そうでなければ loginRequired を返し、HTTP status は 401 を返す。
     * 401 は WWW-Authentication をふくまねばならないので、とりあえず OAuth を入れておく。
     * @return
     */
    @Override
    protected Result doExecute() throws DAOException {
        UserEx user = getLoginUser();
        if (user != null) {
            return renderOK();
        } else {
            return renderLoginRequired();
        }
    }
}

File: app/controllers/api/debug/SuccessIfLoginAPI.scala
package controllers.api.debug

import controllers.api.AbstractAPI
import controllers.ActionContext
import play.api.mvc.Request
import play.api.mvc.AnyContent
import play.api.mvc.PlainResult

/**
 * login していれば success と同じ挙動をする。
 * そうでなければ loginRequired を返し、HTTP status は 401 を返す。
 * 401 は WWW-Authentication をふくまねばならないので、とりあえず OAuth を入れておく。
 */
object SuccessIfLoginAPI extends AbstractAPI[Unit, Unit] {
  override def parseRequest(request: Request[AnyContent])(implicit context: ActionContext): Unit = {
  }

  override def executeAction(params: Unit)(implicit context: ActionContext): Unit = {
  }

  override def renderResult(values: Unit)(implicit context: ActionContext): PlainResult = {
    context.loginUser match {
      case None =>
        return renderLoginRequired()
      case Some(_) =>
        return renderOK()
    }
  }
}
